File: backend/domain/entities/alert.go
package entities

import (
	"time"

	"github.com/google/uuid"
)

type Alert struct {
	ID        uuid.UUID `gorm:"type:uuid;default:gen_random_uuid();primaryKey" json:"id"`
	DeviceID  uuid.UUID `gorm:"type:uuid;not null" json:"device_id"`
	AlertType string    `gorm:"not null" json:"alert_type"` // e.g., "Critical", "Warning"
	Severity  string    `gorm:"not null" json:"severity"`   // e.g., "High", "Medium", "Low"
	Message   string    `gorm:"not null" json:"message"`
	CreatedAt time.Time `json:"created_at"`
	Device    Device    `gorm:"foreignKey:DeviceID" json:"-"`
}

File: mobile_app/lib/features/dashboard/domain/entities/alert.dart
import 'package:equatable/equatable.dart';

class Alert extends Equatable {
  final String id;
  final String deviceId;
  final String severity; // 'Critical', 'Warning', 'Info'
  final String message;
  final DateTime timestamp;
  final bool acknowledged;

  const Alert({
    required this.id,
    required this.deviceId,
    required this.severity,
    required this.message,
    required this.timestamp,
    required this.acknowledged,
  });

  @override
  List<Object?> get props => [id, deviceId, severity, message, timestamp, acknowledged];
}
